src/torchcodec/decoders/_core/CMakeLists.txt

```diff
@@ -3,62 +3,133 @@ project(TorchCodec)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+find_package(pybind11 REQUIRED)
 find_package(Torch REQUIRED)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Werror ${TORCH_CXX_FLAGS}")
 find_package(Python3 ${PYTHON_VERSION} EXACT COMPONENTS Development)
 
-function(make_torchcodec_library library_name ffmpeg_target)
-    set(
-        sources
-        FFMPEGCommon.h
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Werror ${TORCH_CXX_FLAGS}")
+
+function(make_torchcodec_sublibrary
+    library_name
+    type
+    sources
+    library_dependencies)
+
+    add_library(${library_name} ${type} ${sources})
+    set_target_properties(${library_name} PROPERTIES CXX_STANDARD 17)
+    target_include_directories(${library_name}
+        PRIVATE
+        ./../../../../
+        "${TORCH_INSTALL_PREFIX}/include"
+        ${Python3_INCLUDE_DIRS}
+    )
+
+    # Avoid adding the "lib" prefix which we already add explicitly.
+    set_target_properties(${library_name} PROPERTIES PREFIX "")
+
+    target_link_libraries(
+        ${library_name}
+        PUBLIC
+        ${library_dependencies}
+    )
+endfunction()
+
+function(make_torchcodec_libraries
+    ffmpeg_major_version
+    ffmpeg_target)
+
+    # TODO: List each library and its purpose.
+
+    # Create libtorchcodec_decoderN.so
+    set(decoder_library_name "libtorchcodec_decoder${ffmpeg_major_version}")
+    set(decoder_sources
+        AVIOContextHolder.cpp
         FFMPEGCommon.cpp
-        VideoDecoder.h
         VideoDecoder.cpp
-        VideoDecoderOps.h
-        VideoDecoderOps.cpp
-        DeviceInterface.h
     )
+
     if(ENABLE_CUDA)
-        list(APPEND sources CudaDevice.cpp)
+        list(APPEND decoder_sources CudaDevice.cpp)
     else()
-        list(APPEND sources CPUOnlyDevice.cpp)
+        list(APPEND decoder_sources CPUOnlyDevice.cpp)
     endif()
-    add_library(${library_name} SHARED ${sources})
-    set_property(TARGET ${library_name} PROPERTY CXX_STANDARD 17)
 
-    target_include_directories(
-        ${library_name}
-        PRIVATE
-        ./../../../../
-        "${TORCH_INSTALL_PREFIX}/include"
-        ${Python3_INCLUDE_DIRS}
+    set(decoder_library_dependencies
+        ${ffmpeg_target}
+        ${TORCH_LIBRARIES}
```

Comment:
Note that I took out `${Python3_LIBRARIES}` as a dependency. That caused problems with pybind11 importing on Mac: https://github.com/pybind/pybind11/issues/3907#issuecomment-1170412764

Reply:
IIUC the problem wasn't really with `libtorchcodec_decoder` (which shouldn't need Python anyway), it was with the `libtorchcodec_pybind_ops` library - which was pulling `libtorchcodec_decoder`?

Reply:
Yes, because `libtorchcodec_pybind_ops` depends on `libtorchcodec_decoder`, and `libtorchcodec_decoder` used to depend on `${Python3_LIBRARIES}`, we ran into problems at runtime.

Now we're good because `libtorchcodec_decoder` does not depend on `${Python3_LIBRARIES}`, `libtorchcodec_custom_ops` does, and `libtorchcodec_pybind_ops` does not (it gets what it needs through `pybind11::module`).